torch/csrc/autograd/variable.cpp

```diff
@@ -181,6 +200,8 @@ void rebase_history(const Variable& self, Edge gradient_edge) {
   }
 
   set_gradient_edge(self, std::move(gradient_edge));
+  // Pass both self and its grad_fn to avoid calling into grad_fn reentrantly
+  torch::autograd::impl::update_retains_grad(self, self.grad_fn());
```

Comment:
Shouldn't think be done for the view case as well?

Reply:
We discussed this offline already, but just to document it here, we want to have it in the grad_fn instead of rebase history because other views that weren't directly updated in-place will also be affected